### ISSUE TITLE
Sample config file

### ISSUE THREAD

**Issue by logological** [dkpro/dkpro-lsr#8]
As far as I can tell, there is no documentation whatsoever about the config file format for DKPro LSR, nor is a sample file provided in the source tree.  Providing some documentation, or an example file, is critical, as DKPro LSR cannot be used without a config file.

Is it possible there was some documentation about this on Google Code that didn't get migrated here to GitHub?

**Comment by logological** [comment]
Apropos of [this recent query on the DKPro WSD mailing list](https://groups.google.com/forum/#!topic/dkpro-wsd-users/IF-iRGPbiiI), is there any chance this issue could be fixed?  Presumably @zesch has a config file that demonstrates how to configure each of the LSRs supported by DKPro LSR?  (I have a `resources.xml` file but only for WordNet, GermaNet, Wiktionary, and Wikipedia.)

**Comment by zesch** [comment]
There is some documentation in the DKPro Similarity project:
https://dkpro.github.io/dkpro-similarity/settinguptheresources/
and also an example resources.xml
https://public.ukp.informatik.tu-darmstadt.de/dkprosimilarity/resources.xml

Does that help?

-Torsten

﻿On 10.06.19, 17:53, "Tristan Miller" <notifications@github.com> wrote:

    Apropos of 
    this recent query on the DKPro WSD mailing list <https://groups.google.com/forum/#!topic/dkpro-wsd-users/IF-iRGPbiiI>, is there any chance this issue could be fixed? Presumably
    @zesch <https://github.com/zesch> has a config file that demonstrates how to configure each of the LSRs supported by DKPro LSR? (I have a
    resources.xml file but only for WordNet, GermaNet, Wiktionary, and Wikipedia.)
    —
    You are receiving this because you were mentioned.
    Reply to this email directly, 
    view it on GitHub <https://github.com/dkpro/dkpro-lsr/issues/8?email_source=notifications&email_token=AAURBYDCOZUKPGEFHOQUTS3PZZ2GJA5CNFSM4CHPZ62KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXKJC3I#issuecomment-500470125>, or

**Comment by logological** [comment]
Not really—the problem is that different LSRs take different positional arguments, but there is no documentation about them.  The `resources.xml` file you referred to contains beans only for Wiktionary, which takes two arguments, a language ID (looks to be the English-language name of the language, in upper case) and a path to a Wiktionary dump.  Wikipedia resources, on the other hand, take five arguments: a hostname, some sort of code that looks like it might identify an API or a database table, presumably a username and a password, and again some sort of language ID (again, not completely sure what scheme, but looks like another English-language language name, but this time in lower case).  WordNet takes a single argument, which looks like the path to an extJWNL `wordnet_properties.xml` file.

The mailing list enquirer wants to use OpenThesaurus, but I can't even find an example showing what arguments it needs.  The best I'd be able to do is to reverse-engineer the DKPro LSR source code to determine what's required.

Do you happen to have any documentation about which arguments are required for each type of LSR supported by DKPro LSR? Or at least a `resources.xml` file that shows examples for each type of LSR?

**Comment by zesch** [comment]
Ok, I see.

Unfortunately, I don't have a good example.
DKPro LSR is also not really maintained anymore as far as I know.

-Torsten

﻿On 11.06.19, 10:19, "Tristan Miller" <notifications@github.com> wrote:

    Not really—the problem is that different LSRs take different positional arguments, but there is no documentation about them. The
    resources.xml file you referred to contains beans only for Wiktionary, which takes two arguments, a language ID (looks to be the English-language name of the language, in upper case) and a path to a Wiktionary dump. Wikipedia resources, on the
     other hand, take five arguments: a hostname, some sort of code that looks like it might identify an API or a database table, presumably a username and a password, and again some sort of language ID (again, not completely sure what scheme, but looks like another
     English-language language name, but this time in lower case). WordNet takes a single argument, which looks like the path to an extJWNL
    wordnet_properties.xml file.
    The mailing list enquirer wants to use OpenThesaurus, but I can't even find an example showing what arguments it needs. The best I'd be able to do is to reverse-engineer the DKPro LSR source code to determine what's required.
    Do you happen to have any documentation about which arguments are required for each type of LSR supported by DKPro LSR? Or at least a
    resources.xml file that shows examples for each type of LSR?
    —
    You are receiving this because you were mentioned.
    Reply to this email directly, 
    view it on GitHub <https://github.com/dkpro/dkpro-lsr/issues/8?email_source=notifications&email_token=AAURBYBSQWZ5CZNKBHGA4RDPZ5NZDA5CNFSM4CHPZ62KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXMKTDY#issuecomment-500738447>, or

**Comment by reckart** [comment]
This is what I could distill from a `resources.xml` I still had lying around on my machine. No OpenThesaurus in there though...

```
<beans xmlns="http://www.springframework.org/schema/beans"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans-2.5.xsd">
	
	<bean
		class="org.springframework.beans.factory.config.PropertyPlaceholderConfigurer">
	</bean>
	
	<bean id="germanet-example" lazy-init="true"
		class="de.tudarmstadt.ukp.lexsemresource.resource.GermaNetResource">
		<constructor-arg value="${DKPRO_HOME}/de.tudarmstadt.ukp.lexsemresource.util.loader.ResourceLoader/gn_xml/"/>
	</bean>
	
	<bean id="wordnet-example" lazy-init="true"
		class="de.tudarmstadt.ukp.lexsemresource.resource.WordNetResource">
		<constructor-arg value="${DKPRO_HOME}/de.tudarmstadt.ukp.lexsemresource.util.loader.ResourceLoader/wordnet3/wordnet_properties.xml"/>
	</bean>
	
	<bean id="wiktionary-example" lazy-init="true"
		class="de.tudarmstadt.ukp.lexsemresource.resource.WiktionaryResource">
		<constructor-arg value="ENGLISH"/>
		<constructor-arg value="${DKPRO_HOME}/de.tudarmstadt.ukp.lexsemresource.util.loader.ResourceLoader/enwiktionary-20090111__jwktl0.12rc1"/>
	</bean>
	
	<bean id="wikipedia-article-example" lazy-init="true"
		class="de.tudarmstadt.ukp.lexsemresource.resource.WikipediaArticleResource">
		<constructor-arg value="MYSQL HOST"/>
		<constructor-arg value="DBNAME"/>
		<constructor-arg value="USERNAME"/>
		<constructor-arg value="PASSWORD"/>
		<constructor-arg value="LANGUAGE"/>
	</bean>

	<bean id="wikipedia_category-example" lazy-init="true"
		class="de.tudarmstadt.ukp.lexsemresource.resource.WikipediaCategoryResource">
		<constructor-arg value="MYSQL HOST"/>
		<constructor-arg value="DBNAME"/>
		<constructor-arg value="USERNAME"/>
		<constructor-arg value="PASSWORD"/>
		<constructor-arg value="LANGUAGE"/>
	</bean>
</beans>

```

**Comment by logological** [comment]
> DKPro LSR is also not really maintained anymore as far as I know.

Sad if true, as DKPro Similarity depends on it.
